app/src/main/java/com/pps/globant/fittracker/mvp/presenter/WeatherPresenter.java

```diff
@@ -0,0 +1,200 @@
+package com.pps.globant.fittracker.mvp.presenter;
+
+import android.app.Activity;
+import android.content.IntentSender;
+import android.support.annotation.NonNull;
+import android.widget.Toast;
+
+import com.google.android.gms.common.api.ResolvableApiException;
+import com.google.android.gms.location.LocationCallback;
+import com.google.android.gms.location.LocationResult;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.location.LocationSettingsRequest;
+import com.google.android.gms.location.LocationSettingsResponse;
+import com.google.android.gms.location.SettingsClient;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.pps.globant.fittracker.R;
+import com.pps.globant.fittracker.mvp.events.GetWeatherFailureEvent;
+import com.pps.globant.fittracker.mvp.events.GetWeatherSuccessEvent;
+import com.pps.globant.fittracker.mvp.model.WeatherModel;
+import com.pps.globant.fittracker.mvp.view.WeatherView;
+import com.google.android.gms.tasks.Task;
+import com.squareup.otto.Subscribe;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+import static com.pps.globant.fittracker.utils.Constants.REQUEST_CHECK_SETTINGS;
+import static com.pps.globant.fittracker.utils.Constants.WEATHER_LOCATION_PERMISSION_REQUEST_CODE;
+
+public class WeatherPresenter extends LocationCallback {
+
+    private WeatherModel weatherModel;
+    private WeatherView weatherView;
+    private boolean locationPermissions;
+    private boolean locationSettings;
+    private boolean extended = false;
+
+    public WeatherPresenter(WeatherModel weatherModel, WeatherView weatherView) {
+        this.weatherModel = weatherModel;
+        this.weatherView = weatherView;
+    }
+
+    @Subscribe
+    public void onWeatherPermissionsButtonPressed(WeatherView.WeatherPermissionButtonPressedEvent event) {
+        weatherView.toggleSpinner();
+        retrieveLocation();
+    }
+
+    @Subscribe
+    public void onWeatherRefreshButtonPressed(WeatherView.WeatherRefreshButtonPressedEvent event) {
+        weatherModel.retrieveWeather();
+    }
+
+    @Subscribe
+    public void onWeatherSeeMoreButtonPressed(WeatherView.WeatherSeeMoreButtonPressedEvent event) {
+        if (extended) {
+            extended = false;
+            weatherView.hideExtended();
+            return;
+        }
+        fillExtended();
+    }
+
+    private void fillExtended() {
+        // The API returns wind direction in degrees, the array helps translating it into text,
+        // doing a little trick with modulo and division.
+        // Also, the API returns sunset and sunrise times in Epoch time, so it needs a conversion with
+        // the Date and SimpleDateFormat class. As date works in milliseconds, a "*1000" is needed.
+        String directions[] = {"N", "NE", "E", "SE", "S", "SW", "W", "NW", "N"};
```

Comment:
Please move the business logic to the model and only keep the interaction(model -view) in the presenter